java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/ProjectTestUtils.java

```diff
@@ -114,27 +115,59 @@
         return Arrays.stream(entries)
                 .filter(entry -> isTest(entry))
                 .map(entry -> {
-                    final IPath relativePath = entry.getOutputLocation().makeRelativeTo(project.getPath());
+                    final IPath outputLocation = entry.getOutputLocation();
+                    if (outputLocation == null) {
+                        return null;
+                    }
+                    final IPath relativePath = outputLocation.makeRelativeTo(project.getPath());
                     return projectLocation.append(relativePath);
                 })
+                .filter(Objects::nonNull)
                 .collect(Collectors.toSet());
     }
 
-    private static boolean isTest(IClasspathEntry entry) {
+    public static boolean isTest(IJavaProject project, IPath path) {
+        try {
+            final IClasspathEntry entry = project.getClasspathEntryFor(path);
+            if (entry == null) {
+                return false;
+            }
+            return isTest(entry);
+        } catch (final JavaModelException e) {
+            return false;
```

Comment:
any method to log the exception for further diagnose?

Reply:
Good point, I'll solve it in another PR. Tracking by #691